### ISSUE TITLE
Regression: False positive I1101(c-extension-no-member) for math.iclose

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#5081]
### Bug description

[`math.isclose`](https://docs.python.org/3/library/math.html#math.isclose) has been around since 3.5, but now I am getting `c-extension-no-member` for it (I did not see that before).

The same error for `ceil`, `log`, `sqrt` &c

The same for [`resource.getrusage`](https://docs.python.org/3/library/resource.html#resource.getrusage) and [`zlib.crc32`](https://docs.python.org/3/library/zlib.html#zlib.crc32) (and other functions in those modules)

Also, `no-member` for [`dbm.gnu.open`](https://docs.python.org/3/library/dbm.html#dbm.gnu.open) and [`socket.AF_INET`](https://docs.python.org/3/library/socket.html#socket.AF_INET).

File:

```python
"https://github.com/PyCQA/pylint/issues/5081"
import math
math.isclose(1,2)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:3: [I1101(c-extension-no-member), ] Module 'math' has no 'isclose' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects.
```

### Expected behavior

no diagnostics, as before

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.9.7 (default, Sep  3 2021, 12:45:31) 
[Clang 12.0.0 (clang-1200.0.32.29)]
```

**Comment by sam-s** [comment]
Also I get 
```
[E1101(no-member), ...] Instance of 'finfo' has no 'resolution' member
```
for [`np.finfo(np.float64).resolution`](https://numpy.org/doc/stable/reference/generated/numpy.finfo.html) despite `extension-pkg-allow-list=numpy`

**Comment by GoesUp** [comment]
I'm seeing the same issue myself. I'm using pylint 2.10.2 and Python 3.9.7.
